File: mobile-wallet-master/core/src/main/java/org/mifos/mobilewallet/core/base/UseCaseScheduler.java
package org.mifos.mobilewallet.core.base;

/**
 * Interface for schedulers, see {@link UseCaseThreadPoolScheduler}.
 */
public interface UseCaseScheduler {

    void execute(Runnable runnable);

    <V extends UseCase.ResponseValue> void notifyResponse(final V response,
            final UseCase.UseCaseCallback<V>
                    useCaseCallback);

    <V extends UseCase.ResponseValue> void onError(final String message,
            final UseCase.UseCaseCallback<V>
                    useCaseCallback);
}

File: core/data/src/main/java/org/mifos/mobilewallet/core/base/UseCaseScheduler.kt
package org.mifos.mobilewallet.core.base

import org.mifos.mobilewallet.core.base.UseCase.UseCaseCallback

/**
 * Interface for schedulers, see [UseCaseThreadPoolScheduler].
 */
interface UseCaseScheduler {
    fun execute(runnable: Runnable?)
    fun <V : UseCase.ResponseValue?> notifyResponse(
        response: V, useCaseCallback: UseCaseCallback<V>?
    )

    fun <V : UseCase.ResponseValue?> onError(
        message: String?, useCaseCallback: UseCaseCallback<V>?
    )
}
